src/astronomer/flask_appbuilder/security.py

```diff
@@ -201,14 +201,20 @@ def before_request(self):
 
         if current_user.is_anonymous:
             user = self.find_user(username=claims['sub'])
+            name = claims['full_name'] or claims['email']
+            if ' ' in name.strip() and ':' not in name:
+                first_name, last_name = name.split(' ')
+            else:
+                first_name = name
+                last_name = name
             if user is None:
                 log.info('Creating airflow user details for %s from JWT', claims['email'])
                 user = self.user_model(
                     # All we have is REMOTE_USER, so we set
                     # the other fields to blank.
                     username=claims['sub'],
-                    first_name=claims['full_name'] or claims['email'],
-                    last_name='',
+                    first_name=first_name,
+                    last_name=last_name,
```

Comment:
When (which version) did last name start being required? and empty string not working?

Reply:
2.2.0:
https://github.com/apache/airflow/blob/e5da9bf6c985593b9aa624fbb89ed052d871debf/airflow/api_connexion/openapi/v1.yaml#L1851-L1857

Reply:
That's a bug in Airflow that the field is required when FAB doesn't.

Also https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

Reply:
But It won't show to users that those fields are required when they look at the REST API documentation